src/main/scala/com/microsoft/hyperspace/actions/RefreshAction.scala

```diff
@@ -42,12 +42,11 @@ class RefreshAction(
 
   private lazy val previousIndexLogEntry = previousLogEntry.asInstanceOf[IndexLogEntry]
 
-  // Deserialize the plan and create a df.
+  // Reconstruct a df from schema
   private lazy val df = {
-    val serializedPlan = previousIndexLogEntry.source.plan.properties.rawPlan
-    val plan = LogicalPlanSerDeUtils.deserialize(serializedPlan, spark)
-    val qe = spark.sessionState.executePlan(plan)
-    new Dataset[Row](spark, plan, RowEncoder(qe.analyzed.schema))
+    val rel = previousIndexLogEntry.relation
+    val dataSchema = DataType.fromJson(rel.dataSchemaJson).asInstanceOf[StructType]
+    spark.read.schema(dataSchema).format(rel.fileFormat).load(rel.rootPaths: _*)
```

Comment:
What will be the rootPaths in [this](https://github.com/microsoft/hyperspace/pull/99#discussion_r462728758) example?

Reply:
It seems it depends on how the df created:

```
scala> val df2 = spark.read.parquet("/Users/eunjin/repo/spark/bin/testdir/2/b.snappy.parquet","/Users/eunjin/repo/spark/bin/testdir/2/c.snappy.parquet", "/Users/eunjin/repo/spark/bin/testdir/1/a.snappy.parquet")
df2: org.apache.spark.sql.DataFrame = [id: int, name: string ... 1 more field]
==
"rootPaths" : [ "file:/Users/eunjin/repo/spark/bin/testdir/2/b.snappy.parquet", "file:/Users/eunjin/repo/spark/bin/testdir/2/c.snappy.parquet", "file:/Users/eunjin/repo/spark/bin/testdir/1/a.snappy.parquet" ],
```

```
scala> val df2 = spark.read.parquet("/Users/eunjin/repo/spark/bin/testdir/2", "/Users/eunjin/repo/spark/bin/testdir/1/a.snappy.parquet")
df2: org.apache.spark.sql.DataFrame = [id: int, name: string ... 1 more field]
==
 "rootPaths" : [ "file:/Users/eunjin/repo/spark/bin/testdir/2", "file:/Users/eunjin/repo/spark/bin/testdir/1/a.snappy.parquet" ],
"files" : [ "file:/Users/eunjin/repo/spark/bin/testdir/2/b.snappy.parquet", "file:/Users/eunjin/repo/spark/bin/testdir/2/c.snappy.parquet", "file:/Users/eunjin/repo/spark/bin/testdir/1/a.snappy.parquet" ],
```